game_logic.cpp

```diff
@@ -77,39 +79,36 @@ void guess(Board &board, Player &player) {
 	Position guess;
 	int isGuessValid = false; // flag to check if the inputed position is valid. If not, ask the player again
 	while (isGuessValid == false) {
-		std::cout << "Player " << player.id << ", take a guess... " << std::endl;
+		printToPlayer(player, ("Player " + std::to_string(player.id) + ", take a guess... "));

```

Comment:
#must not try to concatenate using a left operand of type `char const*`; create a `std::string` out of it first: `std::string{ "Player " } + std::to_string(player.id) + ", take a guess..."`